common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidDevicePopManager.java

```diff
@@ -203,9 +210,15 @@ private KeyPair generateNewRsaKeyPair(@androidx.annotation.NonNull final Context
                         Logger.error(TAG, "Failed to generate attestation cert - skipping flag.", e);
                         trySetAttestationChallenge = false;
 
+                        continue;
+                    } else if (tryStrongBox && Build.VERSION.RELEASE_OR_CODENAME.equals("UpsideDownCake") && isNegativeInternalError(e.getCause())) {
+                        // Android 14 specific error where strong box is failing, most likely because of IAR requirement in android 14
+                        // https://android.googlesource.com/platform/compatibility/cdd/+/e2fee2f/9_security-model/9_11_keys-and-credentials.md
+                        // Had to check code name, as android 14 device in beta seems to still show 33 as SDK int
+                        Logger.error(TAG, "Android 14 Internal Key store error with strongbox. Skipping strongbox", e);
+                        tryStrongBox = false;
                         continue;
```

Comment:
Let's add work-item link, to investigate the root cause.

Reply:
Sure, done